src/js/faq.js

```diff
@@ -0,0 +1,52 @@
+import Changelog from './changelog';
+import { el, getFullDate } from './utils';
+
+function generateChangelog(anchor) {
+	const root = document.querySelector(`#${anchor} + p`);
+	fetch(Changelog.URL).then(response => response.json()).then(log => {
+		// Reset the content of the changelog root.
+		root.innerHTML = '';
+		const anchors = new Set();
+		const dl = el('dl');
+		log.sort((a, b) => {
+			// Newest changes first.
+			return b.date - a.date;
+		}).forEach(change => {
+			const date = el('dt');
+			date.id = `change-${change.date}`;
+			anchors.add(`#${date.id}`);
+			const fullDate = getFullDate(new Date(change.date));
+			date.innerText = `${fullDate}: ${change.title}`;
+			dl.appendChild(date);
+			const desc = el('dd');
+			// This description may contain URLs.
+			// We are intentionally NOT converting them to clickable anchors
```

Comment:
One secure alternative may be to allow reference URLs in the changelog object. For example:

```
{
  date: 123456,
  title: 'Foo',
  desc: 'Foo bar.',
  url: 'example.com'
}
```

Then here in the generator we can create an anchor tag if `change.url` exists, with the href and innerText values set to that value.

Reply:
👍 

However, if we're already generating the FAQ page on the server, why bother with the client-side stuff here and not also parse and generate the changelog contents in python?

Reply:
Generating the changelog on the server doesn't allay the XSS concerns. We'd still need to decide whether to render the content as text or markup, the only difference being in the jinja template rather than the JS.

Also, the changelog is an external dependency on GitHub, so fetching it in the client at least allows the page to start rendering while it loads. It also avoids the reimplementing the backend load/cache/invalidate logic of an HTTP resource.

Reply:
Fair enough. I like the explicit url pattern. It does limit us to just one URL in the details, but that's probably more than enough — e.g. a pointer to a bug, or discussion elsewhere.

Reply:
Agreed. We can add fanciness if needed, like key/value pairs of label/URL to handle multiple named links. It can appear below the changelog entry as a "See also: X, Y" list of links, kind of similar to how we [link reports with overlapping metrics](https://beta.httparchive.org/reports/state-of-the-web#bytesTotal).

Reply:
Done